finality-aleph/src/block/mod.rs

```diff
@@ -226,3 +232,31 @@ where
     /// Children of the specified block.
     fn children(&self, id: BlockId) -> Result<Vec<J::Header>, Self::Error>;
 }
+
+pub trait HeaderBackend<H: Header>: Send + Sync {
+    type Error: Debug;
+    /// Get block header. Returns `None` if block is not found.
+    fn header(&self, id: BlockId) -> Result<Option<H>, Self::Error>;
+    /// Get hash of a finalized block with a given number. Returns Ok(None) if block exists in
+    /// the database, but is not finalized yet.
+    fn header_of_finalized_at(&self, number: BlockNumber) -> Result<Option<H>, Self::Error>;
+    /// Get currently highest finalized block.
+    fn top_finalized_id(&self) -> BlockId;
+    /// Get full id of a block with a given hash
+    fn hash_to_id(&self, hash: BlockHash) -> Result<Option<BlockId>, Self::Error>;
```

Comment:
`hash_to_header: Hash -> Result<Option<Header>>` would be in a weird conflict with `header: BlockID -> Result<Option<Header>>` and I wanted to discourage users from using the former. Hoped to skip this method completely - it is used exactly once, in the aggregator.

Reply:
It wouldn't be _that_ much of a problem, since in most places you have easier access to a `BlockId` than `Hash`, but feel free to resolve this comment without changing that – it's kinda vibes-based anyway.

Reply:
So I'd rather leave the `hash_to_id`. Re `top_finalized_id` I was slightly inclined to change that to return a header, but that would require some error handling - probably we can unwrap them right away, since the only errors that _should_ possibly appear there are some fatal database errors, but adding that unwrap makes me anxious.